### PR TITLE
Added api documentation to search/filter endpoints

### PR DESCRIPTION
**Pull request by AndreSoderlund**
## Types of changes

- [ ] Bug fix
- [ ] New feature
- [ ] Removed feature
- [ ] Code style update (formatting etc.)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes

## Does this PR introduce a breaking change?
- [x] Yes (I have stepped the version number accordingly)
- [x] No
      
## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly (if applicable).
- [x] I have added/updated tests to cover my changes (if applicable).

### CODE REVIEW

File: src/main/java/se/sundsvall/document/service/Constants.java [hunk 1]
```diff
@@ -16,4 +16,40 @@ private Constants() {}
...
+		- includeConfidential: Should the search include confidential documents? Datatype - boolean (default: false)
+		- onlyLatestRevision: Should the search include only the latest revision of the documents? Datatype - boolean (default: false)
+		- documentTypes: Which document types to include in the search. Datatype - List of Strings
+		- metaData: Uses the metadata object to search for documents with specific metadata. Datatype - List of metadata objects.
+		- page: The page number to retrieve. Datatype - integer (default: 1)
+		- limit: The number of documents to retrieve per page. Datatype - integer (default: 100)
+
+		Objects:
+		- MetaData: {
+			- key: A given metadata key. Datatype - String
```
Comment (josc04) [inline src/main/java/se/sundsvall/document/service/Constants.java:50]: Add that key is optional and that all metadata will be searched if not provided

File: src/main/java/se/sundsvall/document/service/Constants.java [hunk 2]
```diff
@@ -16,4 +16,40 @@ private Constants() {}
...
+			- matchesAny: Returns documents where the given key(if present) have at least one of the matchesAny values. Datatype - List of Strings
+			- matchesAll: Returns documents where the given key(if present) have all of the matchesAll values. Datatype - List of Strings
```
Comment (josc04) [inline src/main/java/se/sundsvall/document/service/Constants.java:52]: Change to `Returns documents where metadata entry with the given key have all of the matchesAll values (if key is present), or if the complete set of metadata have all of the matchesAllValues (when no key is present)`?

File: src/main/java/se/sundsvall/document/service/Constants.java [hunk 3]
```diff
@@ -16,4 +16,40 @@ private Constants() {}
...
+		- onlyLatestRevision: Should the search include only the latest revision of the documents? Datatype - boolean (default: false)
+		- documentTypes: Which document types to include in the search. Datatype - List of Strings
+		- metaData: Uses the metadata object to search for documents with specific metadata. Datatype - List of metadata objects.
+		- page: The page number to retrieve. Datatype - integer (default: 1)
+		- limit: The number of documents to retrieve per page. Datatype - integer (default: 100)
+
+		Objects:
+		- MetaData: {
+			- key: A given metadata key. Datatype - String
+			- matchesAny: Returns documents where the given key(if present) have at least one of the matchesAny values. Datatype - List of Strings
```
Comment (josc04) [inline src/main/java/se/sundsvall/document/service/Constants.java:51]: Change to `Returns documents where metadata entry with the given key have all of the matchesAny values (if key is present), or if the complete set of metadata have all of the matchesAny (when no key is present)`?

**Review by josc04 — APPROVED** [review]

File: src/main/java/se/sundsvall/document/api/DocumentResource.java [hunk 4]
```diff
@@ -225,7 +227,7 @@ ResponseEntity<Void> deleteFile(
 	@GetMapping(produces = {
 		APPLICATION_JSON_VALUE, APPLICATION_PROBLEM_JSON_VALUE
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/document/api/DocumentResource.java:228]: This means HTTP 200 OK can return APPLICATION_PROBLEM_JSON_VALUE

File: src/main/java/se/sundsvall/document/api/DocumentResource.java [hunk 5]
```diff
@@ -241,9 +243,10 @@ ResponseEntity<PagedDocumentResponse> search(
 	@PostMapping(path = "/filter", produces = {
 		APPLICATION_JSON_VALUE, APPLICATION_PROBLEM_JSON_VALUE
```
Comment (cheezi0747) [inline src/main/java/se/sundsvall/document/api/DocumentResource.java]: Missed one

**Review by cheezi0747 — APPROVED** [review]

**Pull request merged**